go/signedexchange/cmd/dump-signedexchange/main.go

```diff
@@ -32,13 +36,45 @@ func run() error {
 
 	if *flagSignature {
 		fmt.Println(e.SignatureHeaderValue)
-	} else {
-		e.PrettyPrint(os.Stdout)
+		return nil
 	}
 
+	e.PrettyPrintHeaders(os.Stdout)
+
+	if *flagVerify {
+		verify(e)
+	}
+
+	e.PrettyPrintPayload(os.Stdout)
+
 	return nil
 }
 
+func verify(e *signedexchange.Exchange) {
```

Comment:
Done.

Since this is a dump tool, verification errors are not considered "hard error", so printing them to stdout and continue processing is fine I think.